server/plugin/plugin_test.go

```diff
@@ -76,52 +88,125 @@ func TestMessageWillBePosted(t *testing.T) {
+			expectedLink: "http://dev.azure.com/abc/xyz/_workitems/edit/1",
+		},
+		{
+			description:  "IsLinkPresent: valid task link 4",
+			msg:          "\n\nhttp://dev.azure.com/abc/xyz/_workitems/edit/1   mock-text",
+			expectedData: []string{"http:", "", "dev.azure.com", "abc", "xyz", "_workitems", "edit", "1"},
+			isValid:      true,
+			regex:        constants.TaskLinkRegex,
+			expectedLink: "http://dev.azure.com/abc/xyz/_workitems/edit/1",
 		},
 		{
-			description: "IsValidTaskLink: invalid link 1",
+			description: "IsLinkPresent: invalid task link 1",
 			msg:         "https://abc/xyz/_workitems/edit/1",
+			regex:       constants.TaskLinkRegex,
 		},
 		{
-			description: "IsValidTaskLink: invalid link 2",
+			description: "IsLinkPresent: invalid task link 2",
 			msg:         "https://dev.azure.com/abc/xyz/_workitems/edit",
+			regex:       constants.TaskLinkRegex,
 		},
 		{
-			description: "IsValidTaskLink: invalid link 3",
+			description: "IsLinkPresent: invalid task link 3",
 			msg:         "https://dev.azure.com/xyz/_workitems/edit/1",
+			regex:       constants.TaskLinkRegex,
 		},
 		{
-			description: "IsValidTaskLink: invalid link 4",
+			description: "IsLinkPresent: invalid task link 4",
 			msg:         "http://dev.azure/abc/xyz/items/it/1",
+			regex:       constants.TaskLinkRegex,
+		},
+		{
+			description:  "IsLinkPresent: valid pull request link 1",
+			msg:          "https://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1/",
+			expectedData: []string{"https:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
+			isValid:      true,
+			expectedLink: "https://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			regex:        constants.PullRequestLinkRegex,
+		},
+		{
+			description:  "IsLinkPresent: valid pull request link 2",
+			msg:          "https://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			expectedData: []string{"https:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
+			isValid:      true,
+			expectedLink: "https://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			regex:        constants.PullRequestLinkRegex,
+		},
+		{
+			description:  "IsLinkPresent: valid pull request link 3",
+			msg:          "http://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			expectedData: []string{"http:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
+			isValid:      true,
+			expectedLink: "http://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			regex:        constants.PullRequestLinkRegex,
+		},
+		{
+			description:  "IsLinkPresent: valid pull request link 4",
+			msg:          "\n\nhttp://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1   mock-text",
+			expectedData: []string{"http:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
+			isValid:      true,
+			expectedLink: "http://dev.azure.com/abc/xyz/_git/xyz/pullrequest/1",
+			regex:        constants.PullRequestLinkRegex,
+		},
```

Comment:
this is good don't remove it now, but if you see ideally we don't need to check these many variations as there could be many possibilities of invalid cases here, so, 1 with valid and 1-2 for invalid would also be acceptable

Reply:
okay